File: app/src/main/java/cm/aptoide/pt/download/FileDownloadTaskStatus.java
package cm.aptoide.pt.download;

import cm.aptoide.pt.downloadmanager.AppDownloadStatus.AppDownloadState;
import cm.aptoide.pt.downloadmanager.DownloadError;
import cm.aptoide.pt.downloadmanager.FileDownloadCallback;
import cm.aptoide.pt.downloadmanager.FileDownloadProgressResult;

/**
 * Created by filipegoncalves on 8/29/18.
 */

public class FileDownloadTaskStatus implements FileDownloadCallback {

  private AppDownloadState appDownloadState;
  private FileDownloadProgressResult downloadProgress;
  private String md5;
  private DownloadError error;
  private int averageDownloadSpeed;
  private int fileType;

  public FileDownloadTaskStatus(AppDownloadState appDownloadState,
      FileDownloadProgressResult downloadProgress, String md5, int averageDownloadSpeed,
      int fileType) {
    this.appDownloadState = appDownloadState;
    this.downloadProgress = downloadProgress;
    this.md5 = md5;
    this.averageDownloadSpeed = averageDownloadSpeed;
    this.fileType = fileType;
    this.error = null;
  }

  public FileDownloadTaskStatus(AppDownloadState appDownloadState, String md5,
      DownloadError error, int averageDownloadSpeed, int fileType) {
    this.appDownloadState = appDownloadState;
    this.md5 = md5;
    this.error = error;
    this.averageDownloadSpeed = averageDownloadSpeed;
    this.fileType = fileType;
    this.downloadProgress = new FileDownloadProgressResult(0, 0);
  }

  @Override public FileDownloadProgressResult getDownloadProgress() {
    return downloadProgress;
  }

  @Override public AppDownloadState getDownloadState() {
    return appDownloadState;
  }

  @Override public String getMd5() {
    return md5;
  }

  @Override public boolean hasError() {
    return error != null;
  }

  @Override public DownloadError getError() {
    return error;
  }

  @Override public int getDownloadSpeed() {
    return averageDownloadSpeed;
  }

  @Override public int getFileType() {
    return fileType;
  }

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    final FileDownloadTaskStatus that = (FileDownloadTaskStatus) o;

    return md5.equals(that.getMd5());
  }
}

File: aptoide-installer/src/main/java/cm/aptoide/pt/aptoide_installer/data/download/filedownloader/FileDownloadTaskStatus.kt
package cm.aptoide.pt.aptoide_installer.data.download.filedownloader

import cm.aptoide.pt.downloadmanager.AppDownloadStatus.AppDownloadState
import cm.aptoide.pt.downloadmanager.FileDownloadCallback
import cm.aptoide.pt.downloadmanager.FileDownloadProgressResult

/**
 * Created by filipegoncalves on 8/29/18.
 */
class FileDownloadTaskStatus : FileDownloadCallback {
  private var appDownloadState: AppDownloadState
  private var downloadProgress: FileDownloadProgressResult
  private var md5: String
  private var error: Throwable?

  constructor(
    appDownloadState: AppDownloadState,
    downloadProgress: FileDownloadProgressResult, md5: String
  ) {
    this.appDownloadState = appDownloadState
    this.downloadProgress = downloadProgress
    this.md5 = md5
    this.error = null
  }

  constructor(appDownloadState: AppDownloadState, md5: String, error: Throwable?) {
    this.appDownloadState = appDownloadState
    this.md5 = md5
    this.error = error
    this.downloadProgress = FileDownloadProgressResult(0, 0)
  }

  override fun getDownloadProgress(): FileDownloadProgressResult {
    return downloadProgress
  }

  override fun getDownloadState(): AppDownloadState {
    return appDownloadState
  }

  override fun getMd5(): String {
    return md5
  }

  override fun hasError(): Boolean {
    return error != null
  }

  override fun getError(): Throwable? {
    return error
  }

  override fun equals(o: Any?): Boolean {
    if (this === o) return true
    if (o == null || javaClass != o.javaClass) return false
    val that = o as FileDownloadTaskStatus
    return md5 == that.getMd5()
  }
}